index.html

```diff
@@ -2435,37 +2435,30 @@ <h2>Cryptographic Suites</h2>
 verification of the data.
         </li>
         <li>
-          The specification MUST define a <dfn>suite selection algorithm</dfn> that accepts a [=JSON
-          object=] |proof| and uses the properties described in [[[#proofs]]] to return either
-          failure or a [=VC cryptographic suite=] [[Infra]] [=struct=]. This algorithm SHOULD be
-          registered in [[[VC-SPECS]]]. A
-          <dfn class="export">VC cryptographic suite</dfn> [=struct=] has the following
-          [=struct/items=].
-          <dl data-dfn-for="cryptographic suite">
-            <dt><dfn>transform</dfn></dt>
-            <dd>A [=transformation algorithm=] that takes a [=JSON object=] as input and returns a
-            [=JSON object=].</dd>
-            <dt><dfn>hash</dfn></dt>
-            <dd>A [=hashing algorithm=] that takes a [=JSON object=] as input and returns a [=byte
-            sequence=].</dd>
-            <dt><dfn>proof creation options</dfn></dt>
+The specification MUST define a <dfn class="export"
+data-lt="cryptosuite instantiation algorithm">data integrity cryptographic
+suite instantiation algorithm</dfn> that accepts a set of options
+([=map=] |options|) and returns a [=cryptosuite instance=]
+([=struct=] |cryptosuite|). This algorithm SHOULD be listed in [[[VC-SPECS]]].
+A <dfn class="export" data-lt="cryptosuite instance">data integrity
+cryptographic suite instance</dfn> [=struct=] has the following
+[=struct/items=].
+          <dl data-dfn-for="cryptosuite instance">
+            <dt><dfn>createProof</dfn></dt>
             <dd>
-              <p>A [=struct=] type that needs to be supplied to the [=cryptographic suite/serialize
-              proof=] operation.</p>
-              <p class="note">To sign data, cryptographic suites need to be able to use a private
-              key. This private key will often be implied by a field in the [=proof creation
-              options=], but it is often a better design to avoid passing the key directly. Passing
-              a key identifier or an asynchronous signing algorithm allows implementations to take
-              advantage of hardware security modules (HSMs) to get better security.</p>
+An algorithm that takes an [=unsecured data document=] ([=map=]
+|unsecuredDocument|) and proof options ([=map=] |options|) as input, and
+produces a [=data integrity proof=] ([=map=]) or an error.
+            </dd>
+            <dt><dfn>verifyProof</dfn></dt>
+            <dd>
+An algorithm that takes a [=secured data document=] ([=map=]
+|securedDocument|) as input, and produces a verification result
+([=map=]) or an error. The verification result MUST contain a
+verification status ([=boolean=] |verified|) value and an [=unsecured data
```

Comment:
This sets up 3 kinds of return values:
1. error
1. non-error with _verified_==false
1. non-error with _verified_==true

If the first two are equivalent, you should collapse them, probably by removing the _verified_ field. If they're different, you should explain what the distinction is.

If the _verified_ field sticks around, you should probably require the _verifiedDocument_ to only be present if it's true?

Reply:
I still haven't done a thorough review here yet, but wanted to note that the first two here aren't equivalent -- so we should explain this a bit more (as @jyasskin says). An error means the verification check couldn't even be completed (because some error prevented it), where as a non-error with `verified=false` means the process was completed and verification was *false*. This might be meaningful in cases where verification could actually become *true* if, for example, network access were restored.

Reply:
Fixed in 1f7ee96701cccc2c1e00992ef61bc397d3bb8388.